### PR TITLE
Add /redact endpoint

### PR DESCRIPTION
**Pull request by Kegsay**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -393,6 +393,14 @@ func (cli *Client) SendText(roomID, text string) (*RespSendEvent, error) {
 		TextMessage{"m.text", text})
 }
 
+// RedactEvent redacts the given event. See http://matrix.org/docs/spec/client_server/r0.2.0.html#put-matrix-client-r0-rooms-roomid-redact-eventid-txnid
+func (cli *Client) RedactEvent(roomID, eventID string, req *ReqRedact) (resp *RespSendEvent, err error) {
+	txnID := "go" + strconv.FormatInt(time.Now().UnixNano(), 10)
```
Comment (NegativeMjark) [inline client.go]: Do we generate txnIDs anywhere else? Could we put the txnID generation code inside a private helper function?

Comment (Kegsay) [inline client.go]: We do generate txnIDs elsewhere. Will make a helper function.

Comment (Kegsay) [inline client.go]: Done.

**Comment by Kegsay** [comment]
@NegativeMjark PTAL

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**